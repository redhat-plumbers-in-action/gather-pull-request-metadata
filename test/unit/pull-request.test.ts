import { describe, it, expect, beforeEach } from 'vitest';

import {
  pullRequestContextFixture,
  IPullRequestTestContext,
} from './fixtures/pull-request.fixture';

describe('Pull Request Object', () => {
  beforeEach<IPullRequestTestContext>(context => {
    context.pullRequests = pullRequestContextFixture.pullRequests;
    context.invalid = pullRequestContextFixture.invalid;
  });

  it<IPullRequestTestContext>('can be instantiated', context =>
    context.pullRequests.map(pullRequestItem =>
      expect(pullRequestItem).toBeDefined()
    ));

  it<IPullRequestTestContext>('getMetadata()', context =>
    context.pullRequests.map(pullRequestItem =>
      expect(pullRequestItem.getMetadata()).toMatchSnapshot()
    ));
});
