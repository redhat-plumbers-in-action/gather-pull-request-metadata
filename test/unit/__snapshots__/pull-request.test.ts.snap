// Vitest Snapshot v1

exports[`Pull Request Object > getMetadata() 1`] = `
{
  "labels": [
    {
      "description": "label-1-description",
      "id": 1,
      "name": "label-1",
    },
  ],
  "milestone": {
    "title": "milestone-1",
  },
  "number": 1,
}
`;

exports[`Pull Request Object > getMetadata() 2`] = `
{
  "labels": [
    {
      "description": undefined,
      "id": undefined,
      "name": undefined,
    },
  ],
  "milestone": {
    "title": undefined,
  },
  "number": 2,
}
`;
